javalib/src/main/scala/java/nio/file/attribute/FileTime.scala

```diff
@@ -21,25 +23,31 @@ final class FileTime private (private val epochMillis: Long,
     }
 
   override def hashCode(): Int =
-    epochMillis.## + nanos.##
+    value.## + unit.##
 
-  def to(unit: TimeUnit): Long =
-    unit.convert(toMillis(), TimeUnit.MILLISECONDS)
+  def to(unit: TimeUnit): Long = unit.convert(value, this.unit)
 
-  def toMillis(): Long = epochMillis
+  def toMillis(): Long = to(TimeUnit.MILLISECONDS)
 
-  def toInstant(): Instant = Instant.ofEpochMilli(epochMillis).plusNanos(nanos)
+  def toInstant(): Instant = {
+    val asNanos = to(TimeUnit.NANOSECONDS)
+    val seconds = to(TimeUnit.SECONDS)
+    Instant.ofEpochSecond(
+      seconds,
+      asNanos - TimeUnit.NANOSECONDS.convert(seconds, TimeUnit.SECONDS))
```

Comment:
Easier and more clearly correct:
```suggestion
      Math.floorMod(asNanos, 1000000000)
```